src/main/java/rx/internal/operators/OnSubscribeCombineLatest.java

```diff
@@ -139,7 +136,7 @@ public void request(long n) {
          * that there is always once who acts on each `tick`. Same concept as used in OperationObserveOn.
          */
         void tick() {
-            if (WIP.getAndIncrement(this) == 0) {
+            if (counter.getAndIncrement() == 0) {
```

Comment:
`counter` should be read into a local variable because the `getAndIncrement` will force the re-read of all fields below, including `counter` itself.